src/jump_input.jl

```diff
@@ -0,0 +1,198 @@
+
+    
+    lower_constraints = Any[]
+    for (F, S) in MOI.get(lower, MOI.ListOfConstraints())
+        for ci in MOI.get(lower, MOI.ListOfConstraintIndices{F,S}())
+            
+            lower_f = MOI.get(lower, MOI.ConstraintFunction(), ci)
+            lower_s = MOI.get(lower, MOI.ConstraintSet(), ci)
+            #@show lower_f, lower_s
+
+            lower_f = MOI.Utilities.map_indices(lower_f) do x
+               return upper_to_model_link[lower_to_upper_link[x]]
+            end
+
+            new_ci = MOI.add_constraint(model, lower_f, lower_s)
+            push!(lower_constraints, new_ci)
+        end
+    end
+ 
+    
+    lower_primal_obj = MOI.get(lower, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}()) 
+    lower_objective = MOI.Utilities.map_indices(lower_primal_obj) do x
+        return upper_to_model_link[lower_to_upper_link[x]]
+    end
+
+    lower_sense = MOI.get(lower, MOI.ObjectiveSense())
+
+    return model, lower_variables, lower_objective, lower_constraints, lower_sense
+end
+
+
+function index_to_row_link(
+    model::MOI.FileFormats.MPS.Model)
+
+    i = 0
+    dict = Dict{MOI.ConstraintIndex, Int}()
+    for (S, _) in MOI.FileFormats.MPS.SET_TYPES
+        for ci in MOI.get(
+            model,
+            MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{Float64}, S}(),
+        )  
+            dict[ci] = i
+            i += 1
+        end
+    end
+    return dict
+end
+
+
+function index_to_column_link(
+    model::MOI.FileFormats.MPS.Model)
+
+    variables = MOI.get(model, MOI.ListOfVariableIndices())
+    return Dict{MOI.VariableIndex,Int}(
+        x => i - 1 for (i, x) in MOI.enumerate(variables)
+    )
+end
+
+
+function write_AUX(
+            )
+
+
+
+#=
+    with open(aux_filename, "w") as OUTPUT:
+    # Num lower-level variables
+    OUTPUT.write("N {}\n".format(len(L.x)))
+    # Num lower-level constraints
+    OUTPUT.write("M {}\n".format(L.b.size))
+    # Indices of lower-level variables
+    nx_upper = len(U.x)
+    for i in range(len(L.x)):
+        OUTPUT.write("LC {}\n".format(i+nx_upper))
+    # Indices of lower-level constraints
+    nc_upper = U.b.size
+    for i in range(L.b.size):
+        OUTPUT.write("LR {}\n".format(i+nc_upper))
+    # Coefficients for lower-level objective
+    for i in range(len(L.x)):
+        OUTPUT.write("LO {}\n".format(L.c[L][i]))
+    # Lower-level objective sense
+    if L.minimize:
+        OUTPUT.write("OS 1\n")
+    else:
+        OUTPUT.write("OS -1\n")
+=#
+
+
+end
+
+
+
+function solve_MibS(
+    model::BilevelModel,
+)
+
+    new_model, lower_variables, lower_objective, lower_constraints, lower_sense  = _build_single_model(model)
+
+    # Write the MPS
+    MOI.write_to_file(new_model, "/Users/hesamshaelaie/Documents/BilevelJuMP.jl/src/model.mps")
+    
+    # Write the AUX
+    io = open("test.txt", "w");
+    println(io, length(lower_variables))
+    println(io, length(lower_constraints))
+
+
+    
+    close(io);
```

Comment:
Instead of 
```Julia
io = open("test.aux", "w")
# ... stuff here
close(io)
```
prefer
```Julia
open("test.aux", "w") do io
    # ... stuff here
end
```

Reply:
got it.